src/clj/ataru/db/migrations.clj

```diff
@@ -426,29 +427,70 @@
 
 (defn- create-attachment-reviews
   [attachment-field application-key hakutoiveet]
-  (let [review-base {:application_key application-key
-                     :attachment_key  (:id attachment-field)
-                     :state           "not-checked"}]
+  (let [review-base                        {:application_key application-key
+                                            :attachment_key  (:id attachment-field)
+                                            :state           "not-checked"}
+        relevant-field-hakukohde-oids      (clojure.set/intersection (set (map :oid hakutoiveet))
+                                                                     (-> attachment-field :belongs-to-hakukohteet set))
+        relevant-field-hakukohderyhma-oids (->> hakutoiveet
+                                                (filter #(not-empty (clojure.set/intersection (-> attachment-field :belongs-to-hakukohderyhma set)
+                                                                                              (-> % :ryhmaliitokset set))))
+                                                (map :oid))
+        hakukohde-oids                     (concat relevant-field-hakukohde-oids relevant-field-hakukohderyhma-oids)]
     (map #(assoc review-base :hakukohde %)
          (cond
-           (not-empty (:belongs-to-hakukohteet attachment-field))
-           (clojure.set/intersection (set hakutoiveet)
-                                     (-> attachment-field :belongs-to-hakukohteet set))
+           (or (-> attachment-field :belongs-to-hakukohderyhma not-empty)
+               (-> attachment-field :belongs-to-hakukohteet not-empty))
+           hakukohde-oids
 
            (not-empty hakutoiveet)
-           hakutoiveet
+           (map :oid hakutoiveet)
 
            :else ["form"]))))
 
+(defn- followup-option-selected?
+  [field answers]
+  (let [parent-answer-key (-> field :followup-of keyword)
+        answers    (-> answers
+                       parent-answer-key
+                       :value
+                       vector ; Make sure we won't flatten a string answer to ()
+                       flatten
+                       set)]
+    (contains? answers (:option-value field))))
+
+(defn- filter-relevant-attachments
+  [answers fields]
+  (filter (fn [field]
+            (and (= "attachment" (:fieldType field))
+                 (or (not (contains? field :followup-of))
+                     (followup-option-selected? field answers))))
+          fields))
+
+(defn- parse-hakukohde
+  [oid hakukohde]
+  {:oid            (or (:oid hakukohde) oid)
+   :ryhmaliitokset (some->> (:ryhmaliitokset hakukohde)
+                            (map #(:ryhmaOid %)))})
+
+(defn- get-hakukohde-rymaliitoket
+  [oid]
+  (parse-hakukohde oid (tarjonta-client/get-hakukohde oid)))
+
+(def get-cached-hakukohde-and-ryhmaliitokset (memoize get-hakukohde-rymaliitoket))
```

Comment:
Tää määrittely kantsii siirtää tuonne `migrate-attachment-states-to-applications` funkkarin sisään. Toi `memoize`:n luoma kakku jää muuten muistiin.

Reply:
Ah totta 😅